recipe-client-addon/test/unit/test_ActionSandboxManager.js

```diff
@@ -2,79 +2,101 @@
     registerAsyncCallback("testCallback", async function(normandy, a, b) {
       return a + b;
     });
-  `, "testCallback", 4, 6);
-  equal(result, 10, "runAsyncCallbackFromScript passes arguments to the callback");
-}));
+  `;
 
-add_task(withManager(async function testCloning(manager) {
-  const result = await manager.runAsyncCallbackFromScript(`
+  await withManager(script, async manager => {
+    const result = await manager.runAsyncCallback("testCallback", 4, 6);
+    equal(result, 10, "runAsyncCallback passes arguments to the callback");
+  });
+});
+
+add_task(async function testCloning() {
+  const script = `
     registerAsyncCallback("testCallback", async function(normandy, obj) {
       return {foo: "bar", baz: obj.baz};
     });
-  `, "testCallback", {baz: "biff"});
-
-  deepEqual(
-    result,
-    {foo: "bar", baz: "biff"},
-    (
-      "runAsyncCallbackFromScript clones arguments into the sandbox and return values into the " +
-      "context it was called from"
-    ),
-  );
-}));
-
-add_task(withManager(async function testError(manager) {
-  try {
-    await manager.runAsyncCallbackFromScript(`
-      registerAsyncCallback("testCallback", async function(normandy) {
-        throw new Error("WHY")
-      });
-    `, "testCallback");
-    ok(false, "runAsnycCallbackFromScript throws errors when raised by the sandbox");
-  } catch (err) {
-    equal(err.message, "WHY", "runAsnycCallbackFromScript clones error messages");
-  }
-}));
-
-add_task(withManager(async function testDriver(manager) {
-  const logInDriver = await manager.runAsyncCallbackFromScript(`
+  `;
+
+  await withManager(script, async manager => {
+    const result = await manager.runAsyncCallback("testCallback", {baz: "biff"});
+
+    deepEqual(
+      result,
+      {foo: "bar", baz: "biff"},
+      (
+        "runAsyncCallback clones arguments into the sandbox and return values into the " +
+        "context it was called from"
+      ),
+    );
+  });
+});
+
+add_task(async function testError() {
+  const script = `
+    registerAsyncCallback("testCallback", async function(normandy) {
+      throw new Error("WHY")
+    });
+  `;
+
+  await withManager(script, async manager => {
+    try {
+      await manager.runAsyncCallback("testCallback");
+      ok(false, "runAsnycCallbackFromScript throws errors when raised by the sandbox");
+    } catch (err) {
+      equal(err.message, "WHY", "runAsnycCallbackFromScript clones error messages");
+    }
+  });
+});
+
+add_task(async function testDriver() {
+  const script = `
     registerAsyncCallback("testCallback", async function(normandy) {
       return "log" in normandy;
     });
-  `, "testCallback");
-  ok(logInDriver, "runAsyncCallbackFromScript passes a driver as the first parameter");
-}));
+  `;
+
+  await withManager(script, async manager => {
+    const logInDriver = await manager.runAsyncCallback("testCallback");
+    ok(logInDriver, "runAsyncCallback passes a driver as the first parameter");
```

Comment:
Only other thing I can think of that would work with the driver (since it's cloned into the sandbox) is some special attribute like `_isNormandyDriver` or something. Thoughts?